torch/testing/_internal/common_utils.py

```diff
@@ -644,16 +644,41 @@ def run_tests(argv=UNITTEST_ARGS):
             failed |= wait_for_process(p) != 0
         assert not failed, "Some test shards have failed"
     elif TEST_SAVE_XML is not None:
-        # import here so that non-CI doesn't need xmlrunner installed
-        import xmlrunner  # type: ignore[import]
-        test_filename = sanitize_test_filename(inspect.getfile(sys._getframe(1)))
-        test_report_path = TEST_SAVE_XML + LOG_SUFFIX
-        test_report_path = os.path.join(test_report_path, test_filename)
-        os.makedirs(test_report_path, exist_ok=True)
-        verbose = '--verbose' in argv or '-v' in argv
-        if verbose:
-            print('Test results will be stored in {}'.format(test_report_path))
-        unittest.main(argv=argv, testRunner=xmlrunner.XMLTestRunner(output=test_report_path, verbosity=2 if verbose else 1))
+
+        def make_xml_runner(argv):
+            test_filename = sanitize_test_filename(inspect.getfile(sys._getframe(1)))
+            test_report_path = TEST_SAVE_XML + LOG_SUFFIX
+            test_report_path = os.path.join(test_report_path, test_filename)
+            os.makedirs(test_report_path, exist_ok=True)
+            verbose = '--verbose' in argv or '-v' in argv
+            if verbose:
+                print('Test results will be stored in {}'.format(test_report_path))
+            # patching library code to add verbosity to test outputs
+            # https://github.com/pytorch/pytorch/issues/69014
+            # import here so that non-CI doesn't need xmlrunner installed
+            # this works with unittest_xml_reporting<=3.2.0,>=2.0.0
+            # (3.2.0 is latest at the moment)
+            from xmlrunner.result import _XMLTestResult  # type: ignore[import]
+
+            def add_skip_with_reason(test_result, test, reason):
+                """
+                Called when a test method was skipped; includes skipping reason
+                """
+                test_result._save_output_data()
```

Comment:
What does this do, just for my curiosity?

Reply:
Looks like it caches current stdout / stderr :
https://github.com/xmlrunner/unittest-xml-reporting/blob/3a2be20177ec6e396d23eecb68d38d410c2ef7a5/xmlrunner/result.py#L400
https://github.com/xmlrunner/unittest-xml-reporting/blob/3a2be20177ec6e396d23eecb68d38d410c2ef7a5/xmlrunner/result.py#L311